template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/oozie/OozieRoleConfigProvider.java

```diff
@@ -13,36 +14,39 @@
 import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
 import com.sequenceiq.cloudbreak.api.endpoint.v4.database.base.DatabaseType;
 import com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRdsRoleConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils;
 import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
 import com.sequenceiq.cloudbreak.template.views.HostgroupView;
 import com.sequenceiq.cloudbreak.template.views.RdsView;
 
 @Component
 public class OozieRoleConfigProvider extends AbstractRdsRoleConfigProvider {
 
-    private static final String OOZIE_DATABASE_HOST = "oozie_database_host";
-
-    private static final String OOZIE_DATABASE_NAME = "oozie_database_name";
-
     private static final String OOZIE_DATABASE_TYPE = "oozie_database_type";
 
     private static final String OOZIE_DATABASE_USER = "oozie_database_user";
 
     private static final String OOZIE_DATABASE_PASSWORD = "oozie_database_password";
 
+    private static final String OOZIE_DATABASE_JDBC_URL = "oozie.service.JPAService.jdbc.url";
+
     private static final String OOZIE_CONFIG_SAFETY_VALVE = "oozie_config_safety_valve";
 
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         switch (roleType) {
             case OozieRoles.OOZIE_SERVER:
                 RdsView oozieRdsView = getRdsView(source);
-                List<ApiClusterTemplateConfig> config = new ArrayList<>(List.of(
-                    config(OOZIE_DATABASE_HOST, oozieRdsView.getHost()),
-                    config(OOZIE_DATABASE_NAME, oozieRdsView.getDatabaseName()),
-                    config(OOZIE_DATABASE_TYPE, oozieRdsView.getSubprotocol()),
-                    config(OOZIE_DATABASE_USER, oozieRdsView.getConnectionUserName()),
-                    config(OOZIE_DATABASE_PASSWORD, oozieRdsView.getConnectionPassword())));
+                StringBuilder ozzieSafetyVale = new StringBuilder();
+                ozzieSafetyVale.append(ConfigUtils
+                        .getSafetyValveProperty(OOZIE_DATABASE_JDBC_URL, oozieRdsView.getConnectionURL()));
```

Comment:
I think the JDBC URL override shall be set conditionally only when the DB SSL is enabled for that `RdsView`. Please see [HMS](https://github.com/hortonworks/cloudbreak/blob/master/template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreConfigProvider.java#L155) and [Ranger Admin](https://github.com/hortonworks/cloudbreak/blob/master/template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java#L152) examples. (The HMS example is more complex as it also supports custom DB properties coming from the custom DH template. The Oozie should like follow the same approach as HMS, i.e. also honor custom blueprint template configs.)

Reply:
On the other hand, we could also go this route and always use the JDBC URL override, no matter what the SSL mode is set to. But then the HMS and Ranger Admin DB configs would have to be also adapted for consistency. I am open to discussions. :smile: